src/pages/Community/CommunityPage.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+import { toast } from 'react-toastify';
+import { TokenInfo } from '@solana/spl-token-registry';
+
+import NonMemberCommunityPage from './NonMemberCommunityPage';
+import NewCommunityPage from './NewCommunityPage';
+import MemberCommunityPage from './MemberCommunityPage';
+import { Icommunity } from '../../common/types';
+import { useTokenRegistry } from '../../common/components/Solana/TokenRegistry';
+import { getCommunityInfo } from '../../common/services/Firebase/GetData/CommunityUtil';
+
+/*
+TODO: check credentials
+*/
+
+export default function CommunityPage(): JSX.Element {
+  const { cId } = useParams<'cId'>();
+  const tokenRegistry = useTokenRegistry();
+
+  const [hasAccess, setHasAccess] = useState(false);
+  const [loadingPage, setLoadingPage] = useState(true);
+
+  const [communityInfo, setCommunityInfo] = useState<Icommunity | undefined>();
+
+  const [tokenInfo, setTokenInfo] = useState<TokenInfo | undefined>();
+  const [tokenRegistryHasLoaded, setTokenRegistryHasLoaded] = useState(false);
+
+  async function handleLoadPage() {
+    setLoadingPage(true);
+    if (cId !== undefined) {
+      try {
+        // check credentials
+        setHasAccess(true);
```

Comment:
Shouldn't hasAccess be set after checking that a token with an mind ID === community ID is present in the users wallet?

Reply:
yeah not done yet
don t have the cloud function to set credentials
it is a TODO

Reply:
well not so easy apparently (top of the file)